src/uniprotkb/components/entry/GoCam.tsx

```diff
@@ -0,0 +1,226 @@
+import { Chip, ExternalLink, Loader } from 'franklin-sites';
+import axios from 'axios';
+import pMap from 'p-map';
+import cn from 'classnames';
+
+import ErrorHandler from '../../../shared/components/error-pages/ErrorHandler';
+import GoCamViz from '../protein-data-views/GoCamViz';
+
+import { useSmallScreen } from '../../../shared/hooks/useMatchMedia';
+import useDataApi from '../../../shared/hooks/useDataApi';
+import useSafeState from '../../../shared/hooks/useSafeState';
+
+import fetchData from '../../../shared/utils/fetchData';
+import { heuristic } from '../../../tools/state/utils/heuristic';
+import * as logging from '../../../shared/utils/logging';
+
+import externalUrls from '../../../shared/config/externalUrls';
+
+import { GoCamModelInfo, GoCamModels } from '../../types/goCamTypes';
+
+import styles from './styles/go-cam.module.scss';
+
+const extractGoCamId = (url: string) => {
+  const reGoCamId = /https?:\/\/model\.geneontology\.org\/(?<goCamId>.*)/;
+  const m = url.match(reGoCamId);
+  return m?.groups?.goCamId;
+};
+
+export const getGoCamStructures = (data: GoCamModels[] = []) => {
+  const idToItem = new Map<string, { id: string; label: string }>();
+  for (const d of data) {
+    const id = extractGoCamId(d.gocam);
+    if (id) {
+      idToItem.set(id, {
+        id,
+        label: d.title,
+      });
+    }
+  }
+  return idToItem;
+};
+
+export const isUniprotCurated = (goCamModel: GoCamModelInfo) =>
+  goCamModel?.annotations?.some(
+    ({ key, value }) =>
+      key === 'providedBy' && value === 'https://www.uniprot.org'
+  );
+
+export const getUniprotNode = (
+  primaryAccession: string,
+  goCamModel: GoCamModelInfo
+) => {
+  const idToFind = `UniProtKB:${primaryAccession}`;
+  for (const individual of goCamModel.individuals) {
+    for (const type of individual.type) {
+      if (type.id === idToFind) {
+        // Exclude the organism eg Hsap
+        return type.label.split(' ').slice(0, -1).join(' ');
+      }
+    }
+  }
+  return null;
+};
+
+type Props = {
+  primaryAccession: string;
+};
+
+const GoCam = ({ primaryAccession }: Props) => {
+  const isSmallScreen = useSmallScreen();
+  const [selectedId, setSelectedId] = useState<string | null>(null);
+  const allGoCamIdsResponse = useDataApi<GoCamModels[]>(
+    isSmallScreen ? null : externalUrls.GeneOntologyModels(primaryAccession)
+  );
+  const [goCamIdToNode, setGoCamIdToNode] = useSafeState<Map<
+    string,
+    string | null
+  > | null>(null);
+
+  const uniprotGoCamIds = goCamIdToNode && Array.from(goCamIdToNode.keys());
+
+  const goCamIdToItem = useMemo(
+    () => getGoCamStructures(allGoCamIdsResponse.data),
+    [allGoCamIdsResponse.data]
+  );
+
+  useEffect(() => {
+    // eslint-disable-next-line import/no-named-as-default-member
+    const cancelTokenSource = axios.CancelToken.source();
+    const abortController = new AbortController();
+    const { signal } = abortController;
+    signal.addEventListener('abort', () => {
```

Comment:
I was wondering why that was needed, but I see the axios cancel token logic above. It's one of these cases where using plain fetch would be easier as AbortController fits well into the native fetch logic.
No need to change here, just a comment